src/core/thread/mle.cpp

```diff
@@ -2508,6 +2508,13 @@ void Mle::HandleUdpReceive(Message &aMessage, const Ip6::MessageInfo &aMessageIn
     neighbor = (command == kCommandChildIdResponse) ? mNeighborTable.FindParent(extAddr)
                                                     : mNeighborTable.FindNeighbor(extAddr);
 
+#if OPENTHREAD_FTD
+    if ((neighbor == nullptr) && (command == kCommandAdvertisement))
```

Comment:
What if we generalize to all messages? If we already have frame counter state, we should probably not accept any replayed messages. Thoughts?
```suggestion
    if (neighbor == nullptr)
```

Reply:
I agree that it feels like the right thing to do but I wonder if there was any reason that we had this behavior in `FindNeighbor()` (where if device is FED we only checked the `Parent` and `ParentCandidate` and excluded the `RxOnlyNeighbors`). 

It may be safer to double-check if setting `aRxInfo.mNeighbor` so it is not `nullptr` and is set to a `RxOnlyNeighbor` on an FED, can impact the processing of any other received MLE message type (we know it should be safe for MLE Advertisements). I will try to check this later.

Reply:
I checked all `Handle{MleMsg}()` methods. I think it should be ok to change this. Here are the ones where we can have a change in behavior (allow processing of message from a `RxOnlyNeighbor`):
- `MleRouter::HandleDataRequest()`
- `Mle::HandleDataResponse()`
- `Mle::HandleLinkProbe()`
- `Mle::HandleLinkMetricsManagementRequest/Response()`

Basically today on FED we won't set `aRxInfo.mNeghbor` if sender is `RxOnlyNeighbor` and so we won't process any of above message from such a sender. If we update the code we would allow these to be processed on FED. To me allowing these seems fine. 

@jwhui what do you think?

-------

I also notcied`Mle::HandleChildIdResponse()`. We have this code which makes the `kCommandChildIdResponse` case special:
```
neighbor = (command == kCommandChildIdResponse) ? mNeighborTable.FindParent(extAddr)
                                                    : mNeighborTable.FindNeighbor(extAddr);
```
We can instead add a check that sender is parent or parent candidate, directly in `HandleChildIdResponse()`.

Reply:
Maybe the only question here is whether we should restrict processing Data Response message to those that are sent by parent. Is there any downside if the FED/REED receives updated Network Data / Operational Datasets before its Parent?

Reply:
Good point! I cannot think of any immediate side-effect but would recommend we lean towards being safe and not allowing it (allow Data Request/Response on FED form parent only). 

Let me submit a new PR with some related changes based on this discussion.

Reply:
Submitted https://github.com/openthread/openthread/pull/9484 replacing this PR.